soc/arm/st_stm32/common/Kconfig.soc

```diff
@@ -8,3 +8,13 @@ DT_CHOSEN_Z_CCM := zephyr,ccm
 
 config STM32_CCM
 	def_bool $(dt_chosen_enabled,$(DT_CHOSEN_Z_CCM))
+
+config STM32_PREINIT
```

Comment:
Please explain why this can't be done using already available mecanisms

Reply:
As mentioned in another comment, we need to do this before the zephyr __start method as the memory for the stack isn't available at that point. We could put the stack in OCRAM and use the FMC only for the HEAP, but that complicates things and I'm not sure how to do that.

Reply:
ok, replied in above comment

Reply:
I  want to add FMC driver too! 
My idea: Adds FMC driver in the z_platform_init() function called at reset.S

Reply:
I tried z_platform_init, but because the stack isn't setup it can't jump back after it's executed the code inside of z_platform_init.